client/src/common/types/entities.ts

```diff
@@ -2,20 +2,33 @@
 
 export type Category = number | string | boolean;
 
+export type ColumnSchema = AnnotationColumn | LayoutColumn | DataFrame;
```

Comment:
Nit on naming -- technically the X matrix (aka dataframe) is not columnar.

These sub-schema objects all describe the schema for a primitive Array or Matrix in one of the fields.  So maybe just call it an ArraySchema?

Reply:
Updated [entities.ts::ArraySchema](https://github.com/chanzuckerberg/cellxgene/blob/mim/annoMatrix/client/src/common/types/schema.ts#L63).